File: OsmAndWear/src/net/osmand/plus/notifications/NotificationDismissReceiver.java
package net.osmand.plus.notifications;

import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import net.osmand.plus.OsmandApplication;
import net.osmand.plus.notifications.OsmandNotification.NotificationType;
import net.osmand.util.Algorithms;

public class NotificationDismissReceiver extends BroadcastReceiver {

	public static final String NOTIFICATION_TYPE_KEY_NAME = "net.osmand.plus.notifications.NotificationType";

	@Override
	public void onReceive(Context context, Intent intent) {
		NotificationHelper helper = ((OsmandApplication) context.getApplicationContext()).getNotificationHelper();
		String notificationTypeStr = intent.getExtras().getString(NOTIFICATION_TYPE_KEY_NAME);
		if (!Algorithms.isEmpty(notificationTypeStr)) {
			try {
				NotificationType notificationType = NotificationType.valueOf(notificationTypeStr);
				helper.onNotificationDismissed(notificationType);
			} catch (Exception e) {
				//ignored
			}
		}
	}

	public static PendingIntent createIntent(Context context, NotificationType notificationType) {
		Intent intent = new Intent(context, NotificationDismissReceiver.class);
		intent.putExtra(NOTIFICATION_TYPE_KEY_NAME, notificationType.name());
		return PendingIntent.getBroadcast(context.getApplicationContext(),
				0, intent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
	}
}

File: OsmAnd-telegram/src/main/java/net/osmand/telegram/notifications/NotificationDismissReceiver.kt
package net.osmand.telegram.notifications

import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.text.TextUtils

import net.osmand.telegram.TelegramApplication
import net.osmand.telegram.notifications.TelegramNotification.NotificationType

class NotificationDismissReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val helper = (context.applicationContext as TelegramApplication).notificationHelper
        val notificationTypeStr = intent.extras!!.getString(NOTIFICATION_TYPE_KEY_NAME)
        if (!TextUtils.isEmpty(notificationTypeStr)) {
            try {
                val notificationType = NotificationType.valueOf(notificationTypeStr)
                helper.onNotificationDismissed(notificationType)
            } catch (e: Exception) {
                //ignored
            }

        }
    }

    companion object {

        const val NOTIFICATION_TYPE_KEY_NAME = "net.osmand.telegram.notifications.NotificationType"

        fun createIntent(context: Context, notificationType: NotificationType): PendingIntent {
            val intent = Intent(context, NotificationDismissReceiver::class.java)
            intent.putExtra(NOTIFICATION_TYPE_KEY_NAME, notificationType.name)
            return PendingIntent.getBroadcast(context.applicationContext,
                    0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
        }
    }
}
